plugins/lookup/spec.py

```diff
@@ -231,14 +229,32 @@ def run(self, terms, variables, **kwargs):
             doctype=doctype,
             keep_tmp_files=keep_tmp_files,
         )
+        if json_schema:
+            if "defaults" in json_schema:
+                defaults = json_schema["defaults"]
+            if "path" in json_schema:
+                schema_out_path = json_schema["path"]
         output["json_skeleton"] = genspec_obj.generate_json_schema(
-            defaults=defaults
+            schema_out_path=schema_out_path, defaults=defaults
         )
+        defaults = False
+        schema_out_path = None
+
+        if xml_schema:
+            if "defaults" in xml_schema:
+                defaults = xml_schema["defaults"]
+            if "path" in xml_schema:
+                schema_out_path = xml_schema["path"]
+            if "annotations" in xml_schema:
+                annotations = xml_schema["annotations"]
         output["xml_skeleton"] = genspec_obj.generate_xml_schema(
-            defaults=defaults, annotations=annotations
+            schema_out_path=schema_out_path, defaults=defaults, annotations=annotations
         )
-        output["tree"] = genspec_obj.generate_tree_schema()
+        schema_out_path = None
+        if tree_schema and "path" in tree_schema:
+            schema_out_path = tree_schema["path"]
+        output["tree"] = genspec_obj.generate_tree_schema(schema_out_path=schema_out_path)
 
         res.append(output)
 
-        return res
+        return output
```

Comment:
The return from lookup plugin is a list and not dict.

Reply:
Updated